ActivatePubKey/handler.ts

```diff
@@ -0,0 +1,216 @@
+  RetrievedLolliPopPubKeys
+} from "../model/lollipop_keys";
+
+import { AssertionWriter, PopDocumentWriter } from "../utils/writers";
+import { PopDocumentReader } from "../utils/readers";
+import { domainErrorToResponseError } from "../utils/errors";
+import {
+  isPendingLollipopPubKey,
+  isValidLollipopPubKey,
+  MASTER_HASH_ALGO,
+  retrievedLollipopKeysToApiActivatedPubKey,
+  getAlgoFromAssertionRef,
+  getAllAssertionsRef
+} from "../utils/lollipopKeys";
+
+export const activatePubKeyForAssertionRef = (
+  popDocumentWriter: PopDocumentWriter,
+  context: Context
+) => (
+  assertionFileName: AssertionFileName,
+  assertionRef: AssertionRef,
+  body: ActivatePubKeyPayload,
+  pubKey: NonEmptyString
+): TE.TaskEither<IResponseErrorInternal, RetrievedLolliPopPubKeys> =>
+  pipe(
+    popDocumentWriter({
+      assertionFileName,
+      assertionRef,
+      assertionType: body.assertion_type,
+      expiredAt: body.expires_at,
+      fiscalCode: body.fiscal_code,
+      pubKey,
+      status: PubKeyStatusEnum.VALID
+    }),
+    TE.mapLeft(error => {
+      const err = error.detail;
+      context.log.error(err);
+      return ResponseErrorInternal(err);
+    })
+  );
+
+// -------------------------------
+// Handler
+// -------------------------------
+
+type ActivatePubKeyHandler = (
+  context: Context,
+  assertion_ref: AssertionRef,
+  body: ActivatePubKeyPayload
+) => Promise<
+  | IResponseSuccessJson<ActivatedPubKey>
+  | IResponseErrorNotFound
+  | IResponseErrorValidation
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+>;
+export const ActivatePubKeyHandler = (
+  popDocumentReader: PopDocumentReader,
+  popDocumentWriter: PopDocumentWriter,
+  assertionWriter: AssertionWriter
+): ActivatePubKeyHandler => (
+  context,
+  assertion_ref,
+  body
+): ReturnType<ActivatePubKeyHandler> =>
+  pipe(
+    popDocumentReader(assertion_ref),
+    TE.mapLeft(domainErrorToResponseError),
+    TE.filterOrElseW(isPendingLollipopPubKey, doc => {
+      const err = `Unexpected status on pop document during activation: ${doc.status}`;
+      context.log.error(err);
+      return ResponseErrorInternal(err);
+    }),
+    TE.bindTo("popDocument"),
+    TE.bindW("assertionFileName", () =>
+      pipe(
+        `${body.fiscal_code}-${assertion_ref}`,
+        AssertionFileName.decode,
+        TE.fromEither,
+        TE.mapLeft(errors => {
+          const err = `Could not decode assertionFileName | ${readableReportSimplified(
+            errors
+          )}`;
+          context.log.error(err);
+          return ResponseErrorInternal(err);
+        }),
+        TE.chainFirst(assertionFileName =>
+          pipe(
+            assertionWriter(assertionFileName, body.assertion),
+            // eslint-disable-next-line sonarjs/no-identical-functions
+            TE.mapLeft(error => {
+              const err = error.detail;
+              context.log.error(err);
+              return ResponseErrorInternal(err);
```

Comment:
Why not extracting this code into an utility instead than disabling lint rules?

Reply:
done in https://github.com/pagopa/io-functions-lollipop/pull/13/commits/b5d5d231fff8ca9e7201d2acf445b0ddaca4a880